modules/gdpr_tasks/src/Anonymizer.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+namespace Drupal\gdpr_tasks;
+
+use Drupal\Component\Annotation\Plugin;
+use Drupal\Component\Plugin\Exception\PluginException;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\TypedData\Exception\ReadOnlyException;
+use Drupal\gdpr_dump\Sanitizer\GdprSanitizerFactory;
+use Drupal\gdpr_fields\GDPRCollector;
+use Drupal\gdpr_tasks\Entity\TaskInterface;
+use Drupal\user\Entity\User;
+
+/**
+ * Anonymizes or removes field values for GDPR.
+ */
+class Anonymizer {
+
+  private $collector;
+
+  private $db;
+
+  private $entityTypeManager;
+
+  private $moduleHandler;
+
+  private $currentUser;
+
+  private $sanitizerFactory;
+
```

Comment:
Can we document the class properties? At least lets stick in the classes so that IDE's can tell what it is we are working with.

Reply:
Similarly the rest of the Anonymizer methods. Can we add @param and @doc definitions to the docblocks?

Reply:
@yanniboi Yes, I can add those. What's the convention for where to apply doc blocks usually? I assumed it was for public/protected members, but not private.

Regarding the IDE, it can already tell without the docblocks on the fields as it picks it up form the type hints on the constructor parameters used to initialize the fields.

Reply:
@JeremySkinner the convention in Drupal is to document everything. If nothing else, it makes it easier for other developers in the team to read and understand what's going on.

Also, generally I would use `protected` rather than private unless there is a specific reason why we'd need this to be the final implementation/use of it. Generally the convention is to assume that people could extend these classes if they required unless there are specific reasons not to (e.g. security).